src/typings.ts

```diff
@@ -420,17 +420,46 @@ export interface Signature {
   concatSig: string
 }
 
-export interface IdentityWalletSchema {
+// WALLET
+export type WalletTypeEthers = 'WalletTypeEthers'
+export type WalletTypeIdentity = 'WalletTypeIdentity'
+export type WalletType = WalletTypeEthers | WalletTypeIdentity
+
+export interface TLWalletSchema {
   TLWalletVersion: number
-  ethersKeystore: string
+  walletType: WalletType
+  meta?: any
+}
+
+interface EthersWalletMeta {
+  walletFromEthers: ethers.Wallet
+}
+
+export interface EthersWalletSchema extends TLWalletSchema {
+  walletType: WalletTypeEthers
+  meta: EthersWalletMeta
+}
+
+interface IdentityWalletMeta extends EthersWalletMeta {
   identityAddress: string
-  walletType: string
 }
 
-export interface EthersWalletSchema {
-  TLWalletVersion: number
-  ethersKeystore: string
-  walletType: string
+export interface IdentityWalletSchema extends TLWalletSchema {
+  walletType: WalletTypeIdentity
+  meta: IdentityWalletMeta
+}
+
+export interface EncryptedTLWalletSchema extends TLWalletSchema {
+  meta: {
+    ethersKeystore: string
+  }
+}
+
+export interface EncryptedIdentityWalletSchema extends TLWalletSchema {
+  meta: {
+    ethersKeystore: string
+    identityAddress: string
```

Comment:
I would remove the identityAddress here, we can just recalculate the address here, because it is now deterministic.